src/java.base/share/classes/jdk/internal/foreign/abi/SharedUtils.java

```diff
@@ -371,7 +371,7 @@ public static VaList emptyVaList() {
     }
 
     static void checkType(Class<?> actualType, Class<?> expectedType) {
-        if (expectedType != actualType) {
+        if (!expectedType.isAssignableFrom(actualType)) {
```

Comment:
This change is needed to accommodate for the fact that UnboxAddress now accepts an Addressable, but a previous binding still outputs either a MemoryAddress or MemorySegment.

Reply:
Thanks for cleaning up these (and corresponding binding classes)